lang/cs/Org.Apache.REEF.IMRU.Tests/TestActiveContextManager.cs

```diff
@@ -0,0 +1,308 @@
+            }
+
+            IList<string> contextIds = new List<string>();
+            contextIds.Add(ContextIdPrefix + "5");
+            var e = new MyFailedEvaluator(contextIds);
+            Action remove = () => activeContextManager.RemovedFailedContextInFailedEvaluator(e);
+            Assert.Throws<IMRUSystemException>(remove);
+        }
+
+        /// <summary>
+        /// Test remove a failed evaluator which has no context associated.
+        /// The scenario may happen when an evaluator failed but context has not created yet. 
+        /// </summary>
+        [Fact]
+        public void TestRemoveFailedEvaluatorWithNoContext()
+        {
+            var activeContextManager = new ActiveContextManager(5);
+            for (int i = 0; i < 5; i++)
+            {
+                var c = new MyActiveContext(ContextIdPrefix + i);
+                activeContextManager.Add(c);
+            }
+
+            var e = new MyFailedEvaluator(null);
+            activeContextManager.RemovedFailedContextInFailedEvaluator(e);
+            Assert.Equal(5, activeContextManager.NumberOfActiveContexts);
+        }
+
+        /// <summary>
+        /// An implementation of IFailedEvaluator for testing
+        /// </summary>
+        private class MyFailedEvaluator : IFailedEvaluator
+        {
+            private readonly IList<string> _contextIds;
+
+            /// <summary>
+            /// Constructor of MyFailedEvaluator
+            /// The contextIds is used in the associated IFailedContext 
+            /// </summary>
+            /// <param name="contextIds"></param>
+            internal MyFailedEvaluator(IList<string> contextIds)
+            {
+                _contextIds = contextIds;
+            }
+
+            public EvaluatorException EvaluatorException
+            {
+                get { throw new NotImplementedException(); }
+            }
+
+            /// <summary>
+            /// Have one failed context in the failed Evaluator
+            /// </summary>
+            public IList<IFailedContext> FailedContexts
+            {
+                get
+                {
+                    if (_contextIds == null)
+                    {
+                        return null;
+                    }
+
+                    IList<IFailedContext> contexts = new List<IFailedContext>();
+                    foreach (var cid in _contextIds)
+                    {
+                        contexts.Add(new MyFailedContext(cid));
+                    }
+                    return contexts;
+                }
+            }
+
+            public Utilities.Optional<Driver.Task.IFailedTask> FailedTask
+            {
+                get { throw new NotImplementedException(); }
+            }
+
+            /// <summary>
+            /// Returns Evaluator id
+            /// </summary>
+            public string Id
+            {
+                get
+                {
+                    if (_contextIds != null && _contextIds.Count == 1)
```

Comment:
Does this logic carry any importance? All uses of IFailedEvaluator.Id in this file are purely for logging purposes. Can we use just one Id for all cases?